modules/order/src/Form/UseMultiProfileConfirmForm.php

```diff
@@ -0,0 +1,390 @@
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getQuestion() {
+    return $this->t('Are you sure you want to switch to using multiple
+      profiles for shipping and billing for the %label order type?', [
+        '%label' => $this->entity->label(),
+      ]
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfirmText() {
+    return $this->t('Switch to Using Multiple Profiles');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCancelUrl() {
+    return $this->entity->toUrl('collection');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    // Check if we have incompatible module versions as modules like Commerce
+    // Shipping and Commerce POS will be affected when we switch to split
+    // profiles.
+    $incompatible_modules = $this->getIncompatibleModules();
+
+    // If we have modules that are running incompatible versions, output a
+    // warning message to the user.
+    if ($incompatible_modules) {
+      $this->messenger()->addWarning($this->t('The following modules are
+        running incompatible versions for the switch to split profiles and it
+        could possibly render the site as unusable.
+        <p><strong>@modules</strong></p>
+        <p>Please upgrade and try again.</p>', [
+          '@modules' => implode('<br>', $incompatible_modules)
+        ]
+      ));
+
+      return;
+    }
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    /** @var \Drupal\commerce_order\Entity\OrderTypeInterface $order_type */
+    $order_type = $this->entity;
+
+    // Create the billing and shipping profile types.
+    $this->createBillingShippingProfiles();
+
+    // Migrate any fields added to the customer profile type to the billing and
+    // shipping profile types.
+    $this->migrateProfileFields();
+
+    // Batch process to migrate the existing order profiles from the customer
+    // profile type to use the billing/shipping profile types.
+    $this->migrateExistingProfiles();
+
+    // Set the useSingleProfile field to FALSE now that we've processed
+    // everything.
+    $order_type->setUseSingleProfile(FALSE);
+    $order_type->save();
+
+    $form_state->setRedirectUrl($order_type->toUrl('collection'));
+  }
+
+  /**
+   * Create the billing and shipping profile types, if not already created.
+   */
+  protected function createBillingShippingProfiles() {
+    $profile_type_storage = $this->entityTypeManager->getStorage('profile_type');
+    /** @var \Drupal\profile\Entity\ProfileTypeInterface $billing_profile_type */
+    $billing_profile_type = $profile_type_storage->load('customer_billing');
+
+    // Import YAML config.
+    $config_path = drupal_get_path('module',
+        'commerce_order') . '/config/shipping_billing_profiles';
```

Comment:
**OPTIONAL CHANGE REQUEST**

For reasons explained earlier I'd rename the folder to `/config/profile_types`. I would also add a README file in there explaining what are these config files doing there instead of being in the install folder.